features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/processor/MobileSegmentExt.kt

```diff
@@ -17,3 +18,23 @@ internal fun MobileSegment.Wireframe.copy(clip: MobileSegment.WireframeClip?): M
         is MobileSegment.Wireframe.WebviewWireframe -> this.copy(clip = clip)
     }
 }
+
+internal fun MobileSegment.Source.Companion.tryFromSource(
+    source: String,
+    internalLogger: InternalLogger
+): MobileSegment.Source {
+    return try {
+        fromJson(source)
+    } catch (e: NoSuchElementException) {
+        internalLogger.log(
+            InternalLogger.Level.ERROR,
+            InternalLogger.Target.USER,
```

Comment:
I am wondering if use can fill the property `DatadogContext.source` himself during the development? if not why we should log it to user?

Reply:
This is just the same approach we are using to convert `source: String` to `source: Enum` in RUM, with logging error to the user. I would agree that it probably makes no sense for the user. In the past it could be fail because of the Proguard, but now we encode string value in the enum directly, so it won't impact), so probably a leftover from the past. 

I will update to the `DEV` target.